tlsconfig/tlsconfig_client.go

```diff
@@ -33,16 +33,24 @@ func (p clientParam) configureClient(cfg *tls.Config) error {
 }
 
 // ClientKeyPairFiles configures the client with the key pair that it should present to servers when communicating using
-// TLS with client authentication (2-way SSL). If this parameter is not provided, the client will not present a
-// certificate.
+// TLS with client authentication (2-way SSL). If neither ClientKeyPairFiles, ClientKeyPair, nor ClientKeyPairProvider
+// are provided, the client will not present a certificate.
 func ClientKeyPairFiles(certFile, keyFile string) ClientParam {
 	return ClientKeyPair(TLSCertFromFiles(certFile, keyFile))
 }
 
-// ClientKeyPair configures the client with the key pair that it should present to servers when communicating using TLS
-// with client authentication (2-way SSL). If this parameter is not provided, the client will not present a certificate.
+// ClientKeyPair configures the client with a static key pair sourced from the TLSCertProvider that it should present to
+// servers when communicating using TLS with client authentication (2-way SSL). If neither ClientKeyPairFiles,
+// ClientKeyPair, nor ClientKeyPairProvider are provided, the client will not present a certificate.
 func ClientKeyPair(certProvider TLSCertProvider) ClientParam {
```

Comment:
Is there a reason to keep the static variant around? Can we just update the logic of this parameter to use `GetCertificate` and then everyone gets it for free?

Are there cases where the supplier function is expensive so we don't want to call it more than we have to?

Reply:
I don't know of any examples, but it's easy to imagine someone could use a non-cached vault based provider or even load from disk. @jlin one idea could be to make the refreshing version the default but also provide a static version? No strong opinions here though

Reply:
Discussed on Slack, and since we can not find any examples of expensive cert providers (rather than returning from memory/cache) we are going to treat this change as a bugfix to make `ClientKeyPair` use `GetCertificate` and not add a separate param. cc @Neverblade

Reply:
Done